pkg/controller/remotemachineset/gcpactuator.go

```diff
@@ -142,3 +168,126 @@ func (a *GCPActuator) getImageID(cd *hivev1.ClusterDeployment, logger log.FieldL
 		return "", errors.New(msg)
 	}
 }
+
+// obtainLease uses the Hive MachinePoolNameLease resource to obtain a unique, single character
+// for use in the name of the machine pool. We are severely restricted on name lengths on GCP
+// and effectively have one character of flexibility with the naming convention originating in
+// the installer.
+func (a *GCPActuator) obtainLease(pool *hivev1.MachinePool, logger log.FieldLogger) (rune, error) {
+	logger.Debugf("obtaining lease for pool: %s", pool.Name)
+
+	cd := &hivev1.ClusterDeployment{}
+	err := a.client.Get(context.TODO(), types.NamespacedName{Namespace: pool.Namespace, Name: pool.Spec.ClusterDeploymentRef.Name}, cd)
+	if err != nil {
+		return 'X', err
+	}
+
+	leases := &hivev1.MachinePoolNameLeaseList{}
+	// TODO: pass ^^ to find command?
+	err = a.client.List(context.TODO(), leases, client.UseListOptions(&client.ListOptions{Namespace: pool.Namespace}))
```

Comment:
Add a label to the lease so that you can limit the list to only leases for the clusterdeployment via the `client.MatchingLabels` function.